include/miroil/miroil/named_cursor.h

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2015 Canonical, Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3, as published by
+ * the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+ * SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef MIROIL_NAMED_CURSOR_H
+#define MIROIL_NAMED_CURSOR_H
+#include <string>
+#include <mir/geometry/size.h>
+#include <mir/graphics/cursor_image.h>
+
+namespace miroil {
+
+/*
+    A hack for storing a named cursor inside a CursorImage, as a way to delegate cursor loading to shell code.
+*/
```

Comment:
I understand why this hack exists in QtMir, but I'm not sure it makes a good API choice for miroil. Not least because it depends upon `mir::graphics::CursorImage` which won't be available to downstream code.